libgadget/exchange.c

```diff
@@ -17,21 +17,36 @@ typedef struct {
 
 static MPI_Datatype MPI_TYPE_PLAN_ENTRY = 0;
 
+/*Small bitfield struct to cache the layout function and particle data*/
+typedef struct {
+    unsigned int ptype : 3;
+    unsigned int target : 8 * sizeof(int) - 3;
+} ExchangePartCache;
+
 typedef struct {
     ExchangePlanEntry * toGo;
     ExchangePlanEntry * toGoOffset;
     ExchangePlanEntry * toGet;
     ExchangePlanEntry * toGetOffset;
     ExchangePlanEntry toGoSum;
     ExchangePlanEntry toGetSum;
+    /*List of particles to exchange*/
+    int * ExchangeList;
+    /*Total number of exchanged particles*/
+    int nexchange;
+    /*First and last particles in this exchange*/
```

Comment:
struct {
  int first;  /* relative to P[0] */
  int last;
} current_batch;

Reply:
Would you be ok with just a comment? Creating a struct for two ints feels like overkill.

Reply:
Yes. Overkill.